src/engine/complex.jl

```diff
@@ -0,0 +1,62 @@
+export SymComplex
+
+struct SymComplex <: Number
```

Comment:
I can't. See https://github.com/JuliaLang/julia/issues/33246

Reply:
I see. Maybe remove this type for now? Maybe number type is enough.

Reply:
I can imagine Julia will not support complex abstract type in near future,

Reply:
I think it is still necessary to have this type since complex number methods can be specialized to this type instead of a more general `Number`. And it actually provides a place to explore whether `AbstractComplex` is necessary or not.